adapters/sovrn/sovrn.go

```diff
@@ -61,6 +39,53 @@ func (s *SovrnAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapt
 		}
 	}
 
+	errs := make([]error, 0, 1)
+	var err error
+
+	for _, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: err.Error(),
+			})
+			return nil, errs
+		}
+
+		var sovrnExt openrtb_ext.ExtImpSovrn
+		if err := json.Unmarshal(bidderExt.Bidder, &sovrnExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: err.Error(),
+			})
+			return nil, errs
+		}
+
+		imp.TagID = getTagid(sovrnExt)
+
+		if imp.BidFloor == 0 && sovrnExt.BidFloor > 0 {
+			imp.BidFloor = sovrnExt.BidFloor
+		}
+
+		// Validate video params if appropriate
+		video := imp.Video
+		if video != nil {
+			if video.MIMEs == nil ||
+				video.MinDuration == 0 ||
+				video.MaxDuration == 0 ||
```

Comment:
Please review this logic again. For video endpoint there is a possibility to have only maxDuration when `requireExactDuration` if false or not specified. In this case the entire request will be rerjected.

Reply:
I'll need to review this with product. I'll let you know how that conversation goes.

Reply:
@cpabst Were you able to check about this with your team?

Reply:
He has been very busy. 😄 
I'd like to keep it this way, and look at this idea as a future enhancement. I see others doing it the same way.